conda/notices/types.py

```diff
@@ -79,5 +80,6 @@ def get_cache_key(cls, url: str, name: str, cache_dir: Path) -> Path:
         """Returns the place where this channel response will be stored as cache"""
         url_obj = parse.urlparse(url)
         path = url_obj.path.replace("/", "-")
-        cache_filename = f"{name}{path}"
+        cache_filename = sanitize_filename(f"{name}{path}")
```

Comment:
What happens if there are collisions? Two different channels with the same package but after sanitization they have the same `cache_filename`?

Is there value in using a hashing algorithm instead of sanitizing?

Reply:
Yeah, I thought of this scenario as well. A hashing algorithm is good idea. This should actually be a pretty trivial change to make.

Reply:
It would only make debugging more difficult later... but I don't think that's a deal breaker.

Reply:
This reminds me of the short discussion we had about using [requests-cache](https://github.com/conda/conda/pull/11462#discussion_r902484853) to reduce the amount of effort to maintain the notices cache system.

I'd suggest to create a SHA256 hash with the provided URL of the channel and not look back, it probably is enough®.

Reply:
I re-implemented this with an md5. This is not a problem right? We don't care about this stuff being visible.

Reply:
☝️ trivial change, so no worries if it's not okay. I wouldn't think that they would ever deprecate/remove this, right?

Reply:
Since we're using SHA256 for many other parts of conda, I'd suggest to just use that instead of MD5. It probably is computationally not more expensive for these short strings than MD5 and consistency works in our favor IMO.